tests/collaborative/collaborative_monkey_party.test.ts

```diff
@@ -0,0 +1,310 @@
+import seedrandom from "seedrandom";
+import { coreTypes, Model } from "../../src";
+import { deepEquals, range, zoneToXc } from "../../src/helpers";
+import {
+  Command,
+  CoreCommand,
+  isPositionDependent,
+  isRangeDependant,
+  isSheetDependent,
+  isTargetDependent,
+  UnboundedZone,
+} from "../../src/types";
+import { getRangeValues } from "../test_helpers";
+import { MockTransportService } from "../__mocks__/transport_service";
+import { setupCollaborativeEnv } from "./collaborative_helpers";
+import { getMonkeyPartyCommands } from "./revisions_party";
+
+/**
+ * The monkey party test simulates random user actions in a collaborative environment
+ * to ensure all users remain synchronized.
+ *
+ * If a randomly generated test fails, it minimizes the failing commands and
+ * generates a minimal test case that you can copy-paste from the console.
+ *
+ * By default, this test runs once with a deterministic seed to prevent
+ * non-deterministic behavior in the CI pipeline.
+ *
+ * It is intended to be run locally from time to time by increasing `PARTY_COUNT`.
+ */
+
+const PARTY_COUNT = 1; // increase this number to run more tests and enjoy the parties 🐵🎉
+
+describe("monkey party", () => {
+  const now = Date.now();
+  const seeds = range(0, PARTY_COUNT).map((i) => (now + i).toString());
+
+  test.each(PARTY_COUNT === 1 ? ["deterministic-seed"] : seeds)(
+    "monkey party with seed %s",
+    (seed) => {
+      seedrandom(seed, { global: true });
+      const { network, alice, bob, charlie } = setupCollaborativeEnv();
+      const commands = getMonkeyPartyCommands();
+      addUndoRedo(commands);
+      randomizeCommandsPayload(commands);
+
+      const actions = assignUser(shuffle(commands), [alice, bob, charlie]);
+      const concurrentActions = randomizeConcurrentActions(actions);
+      const { fail, executedActions } = run(network, [alice, bob, charlie], concurrentActions);
+      if (fail) {
+        const generatedTestCode = actionsToTestCode(
+          `failed with seed ${seed}`,
+          minimizeFailingCommands(executedActions)
+        );
+        console.log(generatedTestCode);
+        throw new Error(`Failed with seed ${seed}!`);
+      }
+      expect([alice, bob, charlie]).toHaveSynchronizedExportedData();
```

Comment:
this is already done in `run`, no ? or are `toHaveSynchronizedExportedData()` and `areSynced()` different ?

Reply:
Yes it's the same, but this prints a nice jest diff in case of failure, while `run` does not